### PR TITLE
Make Scaffold.of() more helpful when it can't find a Scaffold

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -425,7 +425,35 @@ class Scaffold extends StatefulWidget {
...
+  /// To return null if there is no [Scaffold], then pass `nullOk: true`.
+  static ScaffoldState of(BuildContext context, { bool nullOk: false }) {
+    assert(nullOk != null);
+    assert(context != null);
+    ScaffoldState result = context.ancestorStateOfType(const TypeMatcher<ScaffoldState>());
+    if (nullOk || result != null)
+      return result;
+    throw new FlutterError(
+      'Scaffold.of() called with a context that does not contain a Scaffold.\n'
+      'No Scaffold could be found starting from the context that was passed to Scaffold.of(). '
```
Comment (abarth) [inline packages/flutter/lib/src/material/scaffold.dart]: I might say something about "upward" here.  For example: "No Scaffold ancestor could be found starting from the context that ..."

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/scaffold.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/scaffold.dart]: (Will fix in subsequent patch, since moving a branch to another development machine is a pain.)

Comment (sethladd) [inline packages/flutter/lib/src/material/scaffold.dart]: Agree re: indicating a direction in which we looked :)

**Pull request merged**